// ng-katex/esm5/lib/ng-katex.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, ElementRef, EventEmitter, Input, Output, } from '@angular/core';
import { KatexService } from './ng-katex.service';
import * as ko from './ng-katex.options';
var KatexDirective = /** @class */ (function () {
    function KatexDirective(el, katexService) {
        this.el = el;
        this.katexService = katexService;
        this.onError = new EventEmitter();
    }
    /**
     * @return {?}
     */
    KatexDirective.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        try {
            this.katexService.render(this.equation, this.el, this.options);
        }
        catch (e) {
            this.onError.emit(e);
        }
    };
    KatexDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[katex]',
                },] }
    ];
    /** @nocollapse */
    KatexDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: KatexService }
    ]; };
    KatexDirective.propDecorators = {
        equation: [{ type: Input, args: ['katex',] }],
        options: [{ type: Input, args: ['katex-options',] }],
        onError: [{ type: Output }]
    };
    return KatexDirective;
}());
export { KatexDirective };
if (false) {
    /** @type {?} */
    KatexDirective.prototype.equation;
    /** @type {?} */
    KatexDirective.prototype.options;
    /** @type {?} */
    KatexDirective.prototype.onError;
    /**
     * @type {?}
     * @private
     */
    KatexDirective.prototype.el;
    /**
     * @type {?}
     * @private
     */
    KatexDirective.prototype.katexService;
}
