// ng-katex/esm5/lib/ng-katex.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output, } from '@angular/core';
import * as ko from './ng-katex.options';
var KatexComponent = /** @class */ (function () {
    function KatexComponent() {
        this.onError = new EventEmitter();
    }
    /**
     * @param {?} error
     * @return {?}
     */
    KatexComponent.prototype.hasError = /**
     * @param {?} error
     * @return {?}
     */
    function (error) {
        this.onError.emit(error);
    };
    KatexComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-katex',
                    template: "\n    <span [katex]=\"equation\"\n          [katex-options]=\"options\"\n          (someEvent)=\"hasError($event)\">\n    </span>\n  "
                }] }
    ];
    KatexComponent.propDecorators = {
        equation: [{ type: Input }],
        options: [{ type: Input }],
        onError: [{ type: Output }]
    };
    return KatexComponent;
}());
export { KatexComponent };
if (false) {
    /** @type {?} */
    KatexComponent.prototype.equation;
    /** @type {?} */
    KatexComponent.prototype.options;
    /** @type {?} */
    KatexComponent.prototype.onError;
}
