// ng-katex/esm5/lib/ng-katex-html.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { extractMath } from 'extract-math';
import { DomSanitizer } from '@angular/platform-browser';
import { KatexService } from './ng-katex.service';
var KatexHtmlComponent = /** @class */ (function () {
    function KatexHtmlComponent(domSanitizer, katexService) {
        this.domSanitizer = domSanitizer;
        this.katexService = katexService;
    }
    Object.defineProperty(KatexHtmlComponent.prototype, "allHtml", {
        get: /**
         * @return {?}
         */
        function () {
            var _this = this;
            /** @type {?} */
            var segments = this.segments;
            /** @type {?} */
            var allHtml = segments.map((/**
             * @param {?} seg
             * @return {?}
             */
            function (seg) {
                if (seg.math) {
                    return _this.katexService.renderToString(seg.raw, { displayMode: seg.type === 'display' });
                }
                else {
                    return seg.value;
                }
            })).reduce((/**
             * @param {?} total
             * @param {?} current
             * @return {?}
             */
            function (total, current) {
                return total += current;
            }));
            return this.domSanitizer.bypassSecurityTrustHtml(allHtml);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(KatexHtmlComponent.prototype, "segments", {
        get: /**
         * @return {?}
         */
        function () {
            return extractMath(this.html);
        },
        enumerable: true,
        configurable: true
    });
    KatexHtmlComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-katex-html',
                    template: "<span [innerHTML]=\"allHtml\"></span>"
                }] }
    ];
    /** @nocollapse */
    KatexHtmlComponent.ctorParameters = function () { return [
        { type: DomSanitizer },
        { type: KatexService }
    ]; };
    KatexHtmlComponent.propDecorators = {
        html: [{ type: Input }]
    };
    return KatexHtmlComponent;
}());
export { KatexHtmlComponent };
if (false) {
    /** @type {?} */
    KatexHtmlComponent.prototype.html;
    /**
     * @type {?}
     * @private
     */
    KatexHtmlComponent.prototype.domSanitizer;
    /**
     * @type {?}
     * @private
     */
    KatexHtmlComponent.prototype.katexService;
}
