// ng-katex/esm2015/lib/ng-katex.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, ElementRef, EventEmitter, Input, Output, } from '@angular/core';
import { KatexService } from './ng-katex.service';
import * as ko from './ng-katex.options';
export class KatexDirective {
    /**
     * @param {?} el
     * @param {?} katexService
     */
    constructor(el, katexService) {
        this.el = el;
        this.katexService = katexService;
        this.onError = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        try {
            this.katexService.render(this.equation, this.el, this.options);
        }
        catch (e) {
            this.onError.emit(e);
        }
    }
}
KatexDirective.decorators = [
    { type: Directive, args: [{
                selector: '[katex]',
            },] }
];
/** @nocollapse */
KatexDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: KatexService }
];
KatexDirective.propDecorators = {
    equation: [{ type: Input, args: ['katex',] }],
    options: [{ type: Input, args: ['katex-options',] }],
    onError: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    KatexDirective.prototype.equation;
    /** @type {?} */
    KatexDirective.prototype.options;
    /** @type {?} */
    KatexDirective.prototype.onError;
    /**
     * @type {?}
     * @private
     */
    KatexDirective.prototype.el;
    /**
     * @type {?}
     * @private
     */
    KatexDirective.prototype.katexService;
}
